contracts/core/OperationExecutor.sol

```diff
@@ -120,16 +126,74 @@ contract OperationExecutor is IERC3156FlashBorrower {
     bytes calldata data
   ) external override returns (bytes32) {
     address lender = registry.getRegisteredService(FLASH_MINT_MODULE);
+    FlashloanData memory flData = abi.decode(data, (FlashloanData));
+
+    checkIfLenderIsTrusted(lender);
+    checkIfFlashloanedAssetIsTheRequiredOne(asset, flData.asset);
+    checkIfFlashloanedAmountIsTheRequiredOne(asset, flData.amount);
```

Comment:
```suggestion
    checkIfFlashloanedAmountIsTheRequiredOne(amount, flData.amount);
```

Reply:
`asset` is the correct  one. WE pass the asset to check the contract balanace and see if's more than the required one ( `flData.amount `

Reply:
Right got it!!